cairo/ethereum/cancun/trie.cairo

```diff
@@ -846,6 +896,505 @@ func bytes_to_nibble_list{bitwise_ptr: BitwiseBuiltin*}(bytes_: Bytes) -> Bytes
+    _prepare_trie_inner_account(
+        account_trie, account_trie.value._data.value.dict_ptr_start, mapping_ptr_start
+    );
+    jmp end;
+
+    storage:
+    let storage_trie = trie_union.value.storage;
+    _prepare_trie_inner_storage(
+        storage_trie, storage_trie.value._data.value.dict_ptr_start, mapping_ptr_start
+    );
+    jmp end;
+
+    transaction:
+    let transaction_trie = trie_union.value.transaction;
+    _prepare_trie_inner_transaction(
+        transaction_trie, transaction_trie.value._data.value.dict_ptr_start, mapping_ptr_start
+    );
+    jmp end;
+
+    receipt:
+    let receipt_trie = trie_union.value.receipt;
+    _prepare_trie_inner_receipt(
+        receipt_trie, receipt_trie.value._data.value.dict_ptr_start, mapping_ptr_start
+    );
+    jmp end;
+
+    withdrawal:
+    let withdrawal_trie = trie_union.value.withdrawal;
+    _prepare_trie_inner_withdrawal(
+        withdrawal_trie, withdrawal_trie.value._data.value.dict_ptr_start, mapping_ptr_start
+    );
+    jmp end;
+
+    end:
+    let range_check_ptr = [ap - 5];
+    let bitwise_ptr = cast([ap - 4], BitwiseBuiltin*);
+    let keccak_ptr = cast([ap - 3], KeccakBuiltin*);
+    let poseidon_ptr = cast([ap - 2], PoseidonBuiltin*);
+    let mapping_ptr_end = cast([ap - 1], BytesBytesDictAccess*);
+
+    tempvar result = MappingBytesBytes(
+        new MappingBytesBytesStruct(
+            cast(mapping_ptr_start, BytesBytesDictAccess*),
+            cast(mapping_ptr_end, BytesBytesDictAccess*),
+            cast(0, MappingBytesBytesStruct*),
+        ),
+    );
+    return result;
+}
+
+func _prepare_trie_inner_account{
+    range_check_ptr,
+    bitwise_ptr: BitwiseBuiltin*,
+    keccak_ptr: KeccakBuiltin*,
+    poseidon_ptr: PoseidonBuiltin*,
+}(
+    trie: TrieAddressOptionalAccount,
+    dict_ptr: AddressAccountDictAccess*,
+    mapping_ptr_end: BytesBytesDictAccess*,
+) -> BytesBytesDictAccess* {
+    alloc_locals;
+
+    if (dict_ptr == trie.value._data.value.dict_ptr) {
+        return mapping_ptr_end;
+    }
+
+    // Skip all None values
+    if (cast(dict_ptr.new_value.value, felt) == 0) {
+        return _prepare_trie_inner_account(
+            trie, dict_ptr + AddressAccountDictAccess.SIZE, mapping_ptr_end
+        );
+    }
+
+    let preimage = Bytes20_to_Bytes(dict_ptr.key);
+    let value = dict_ptr.new_value;
+    // TODO: get storage root
+    let (buffer: felt*) = alloc();
+    tempvar storage_root = Bytes(new BytesStruct(buffer, 0));
+    tempvar node = Node(
+        new NodeEnum(
+            account=value,
+            bytes=Bytes(cast(0, BytesStruct*)),
+            legacy_transaction=LegacyTransaction(cast(0, LegacyTransactionStruct*)),
+            receipt=Receipt(cast(0, ReceiptStruct*)),
+            uint=cast(0, Uint*),
+            u256=U256(cast(0, U256Struct*)),
+            withdrawal=Withdrawal(cast(0, WithdrawalStruct*)),
+        ),
+    );
+    let encoded_value = encode_node(node, storage_root);
```

Comment:
why not just using `encode_account` directly?

Reply:
ok why not keeping the encode_node everywhere

Reply:
EELS uses encode_node, I don't mind honestly, we already know what the type used is. it makes more sense.

Reply:
and there's a tiny difference for account where we check the storage root is not None